### PR TITLE
Refactor Member Ignoring Method smell in TaskVisualizer class

### PR DESCRIPTION
**Pull request by emaiannone**
Hi, I'm Emanuele Iannone, a master student at University of Salerno.
Since my bachelor's thesis I have been working on a **code smell refactoring plugin** called *aDoctor*, which is able to identify and fix energy-related problems in Android apps.
I launched it on your project, finding different instances of code smells. I chose one of them and let the plugin automatically fix it.
In this case I chose Member Ignoring Method, that is present when a non static method does not use at all instance variables and other non static methods. These kind of smell may have a non trivial impact on energy consumption, as shown in this paper: `https://www.sciencedirect.com/science/article/pii/S0950584918301678`.
Besides, this kind of refactoring does not impact on the functionalities of your app, so it is totally safe. Let me know if you are interested in this refactoring proposal.

### CODE REVIEW

**Comment by yvolk** [comment]
Hello @emaiannone 
Thank you for noticing the problem. Due to the high volume of changes sometimes we miss such things... Actually due to more attention to functional programming static methods in general become more preferable than instance methods...

Regarding code smell refactoring plugin, is this plugin for IntelliJ Idea / Android Studio?

Generally I use Sonar Cloud for static code analysis and fix automatically found issues later. E.g. for that project: https://sonarcloud.io/dashboard?id=andstatus
I will add ToDo Agenda there also... It would be much more quicker to see and be able to fix such simple problems in the IDE, without switching between Web and IDE...

Regarding this Pull Request:
1. It should be done in that repository: https://github.com/andstatus/todoagenda , not here.
2. Actual change is addition of "static " to one line, all other changes are a bug of your plugin :-) and should not be done IMHO.
3. It would be ideal to make all (or many...) fixes in one commit, not a single commit for one word change :-) Is this possible?

File: app/src/main/java/org/andstatus/todoagenda/task/TaskVisualizer.java [hunk 1]
```diff
@@ -100,12 +100,12 @@ public int getViewTypeCount() {
...
 
-    private List<TaskEntry> createEntryList(List<TaskEvent> events) {
-        List<TaskEntry> entries = new ArrayList<>();
-        for (TaskEvent event : events) {
-            entries.add(TaskEntry.fromEvent(event));
-        }
-        return entries;
-    }
+    private static List<TaskEntry> createEntryList(List<TaskEvent> events) {
+		List<TaskEntry> entries = new ArrayList<>();
```
Comment (yvolk) [inline app/src/main/java/org/andstatus/todoagenda/task/TaskVisualizer.java]: Bug: Formatting changed!

**Comment by yvolk** [comment]
@emaiannone And I see another bug in your PR / plugin:
when you change a method to "static" you should change a way it's referenced...

**Comment by emaiannone** [comment]
Hello @yvolk and thank you for your feedback.

The plugin is available in JetBrains Plugin Repository: feel free to try it. Please note that it is still under development, so it has some problems to be fixed, for example the formatting issue as you mentioned above.

I did not know I had to go on the other repo, sorry :(

Right now, being under development, I have not implemented a "fix all smells" feature: it is on my checklist :)

**Comment by yvolk** [comment]
> Right now, being under development, I have not implemented a "fix all smells" feature: it is on my checklist :) In this case I only wanted to show a sample of its capabilities, that's why I chose only a single instance of MIM smell.

I would say that searching for "smells" and allowing to fix found occurrences with minimal manual intervention should be the main goal of such plugins. Addition of one word to the place that you already know is not a big deal :-)

**Comment by emaiannone** [comment]
Thank you @yvolk, I might try to implement a "fix all" feature or, alternatively, a "lint checker" mode.

**Pull request closed without merging**